vCenter Receiver : tls settings for vCenter receiver doesn't work as expected.

**Describe the bug**

vCenter receiver throws an error even after the tls settings are set as follows(Please note that this is only for verification, we would have tls settings enabled in Prod Environments)
**tls:
      insecure: true
      insecure_skip_verify: true**

Strange thing is that, this error gets reported after some amount of time and the the only way so far to recover is to kill the docker and restart another one.

Can someone help on how to look at this issue ?
otel-collector_1  | 2022-08-20T06:26:18.532Z    error   scraperhelper/scrapercontroller.go:197  Error scraping metrics  {"kind": "receiver", "name": "vcenter", "error": "unable to get datacenter lists: Post \"https://172.16.5.13/sdk\": x509: cannot validate certificate for 172.16.5.13 because it doesn't contain any IP SANs", "scraper": "vcenter"}
otel-collector_1  | go.opentelemetry.io/collector/receiver/scraperhelper.(*controller).scrapeMetricsAndReport
otel-collector_1  |     go.opentelemetry.io/collector@v0.53.0/receiver/scraperhelper/scrapercontroller.go:197
otel-collector_1  | go.opentelemetry.io/collector/receiver/scraperhelper.(*controller).startScraping.func1
otel-collector_1  |     go.opentelemetry.io/collector@v0.53.0/receiver/scraperhelper/scrapercontroller.go:172
otel-collector_1  | 2022-08-20T06:26:18.532Z    INFO    loggingexporter/logging_exporter.go:57  MetricsExporter {"#metrics": 0}
otel-collector_1  | 2022-08-20T06:26:18.532Z    DEBUG   loggingexporter/logging_exporter.go:67
otel-collector_1  | 2022-08-20T06:26:18.533Z    info    exporterhelper/queued_retry.go:215      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "name": "prometheusremotewrite", "error": "invalid tsMap: cannot be empty map", "interval": "55.233015ms"}
otel-collector_1  | 2022-08-20T06:26:28.591Z    info    exporterhelper/queued_retry.go:215      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "name": "prometheusremotewrite", "error": "invalid tsMap: cannot be empty map", "interval": "114.858199ms"}
otel-collector_1  | 2022-08-20T06:26:38.779Z    info    exporterhelper/queued_retry.go:215      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "name": "prometheusremotewrite", "error": "invalid tsMap: cannot be empty map", "interval": "245.836146ms"}
otel-collector_1  | 2022-08-20T06:26:48.802Z    info    exporterhelper/queued_retry.go:215      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "name": "prometheusremotewrite", "error": "invalid tsMap: cannot be empty map", "interval": "259.441708ms"}
otel-collector_1  | 2022-08-20T06:26:58.873Z    info    exporterhelper/queued_retry.go:215      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "name": "prometheusremotewrite", "error": "invalid tsMap: cannot be empty map", "interval": "289.038953ms"}
otel-collector_1  | 2022-08-20T06:27:08.907Z    info    exporterhelper/queued_retry.go:215      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "name": "prometheusremotewrite", "error": "invalid tsMap: cannot be empty map", "interval": "291.204126ms"}
otel-collector_1  | 2022-08-20T06:27:18.462Z    error   exporterhelper/queued_retry_inmemory.go:107     Exporting failed. No more retries left. Dropping data.  {"kind": "exporter", "name": "prometheusremotewrite", "error": "max elapsed time expired invalid tsMap: cannot be empty map", "dropped_items": 0}
otel-collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper.onTemporaryFailure
otel-collector_1  |     go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/queued_retry_inmemory.go:107
otel-collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send
otel-collector_1  |     go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/queued_retry.go:199
otel-collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsSenderWithObservability).send
otel-collector_1  |     go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/metrics.go:132
otel-collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1
otel-collector_1  |     go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/queued_retry_inmemory.go:119
otel-collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper/internal.consumerFunc.consume
otel-collector_1  |     go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/internal/bounded_memory_queue.go:82
otel-collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers.func2
otel-collector_1  |     go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/internal/bounded_memory_queue.go:69


**Steps to reproduce**
start the docker using docker-composer up using the steps mentioned here.
https://opentelemetry.io/docs/collector/getting-started/

**What did you expect to see?**
Issue is that thought we have skipped tls settings verification, still we get the error reporting unauth certificate. We expect that it should skip the cert verification

**What did you see instead?**

**What version did you use?**
Version: 0.53.0

**What config did you use?**
Config:
[root@testotel gl-gateway-opentelemetry-collector]# cat /etc/otel-collector-config.yaml
receivers:
  vcenter:
    endpoint: "https://<vcenter_ip>"
    username: '<username>'
    password: '<password>'
    collection_interval: 1m
    tls:
      insecure: true
      insecure_skip_verify: true

exporters:
  logging:
    loglevel: debug
  prometheusremotewrite:
    endpoint: "https://<end-Point>"
    headers:
      Authorization: "Bearer <token>"

service:
  pipelines:
    metrics:
      receivers: [vcenter]
      exporters: [logging, prometheusremotewrite]


**Environment**
OS: Debian

**Additional context**
Add any other context about the problem here.

--- a/receiver/vcenterreceiver/client.go
+++ b/receiver/vcenterreceiver/client.go
@@ -47,7 +47,10 @@ func newVcenterClient(c *Config) *vcenterClient {
 // EnsureConnection will establish a connection to the vSphere SDK if not already established
 func (vc *vcenterClient) EnsureConnection(ctx context.Context) error {
 	if vc.moClient != nil {
-		return nil
+		sessionActive, _ := vc.moClient.SessionManager.SessionIsActive(ctx)
+		if sessionActive {
+			return nil
+		}
 	}
 
 	sdkURL, err := vc.cfg.SDKUrl()

--- a/receiver/vcenterreceiver/client_test.go
+++ b/receiver/vcenterreceiver/client_test.go
@@ -16,12 +16,16 @@ package vcenterreceiver // import github.com/open-telemetry/opentelemetry-collec
 
 import (
 	"context"
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/require"
+	"github.com/vmware/govmomi"
 	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/session"
 	"github.com/vmware/govmomi/simulator"
 	"github.com/vmware/govmomi/vim25"
+	"go.opentelemetry.io/collector/config/configtls"
 )
 
 func TestGetClusters(t *testing.T) {
@@ -62,5 +66,35 @@ func TestGetVMs(t *testing.T) {
 		vms, err := client.VMs(ctx)
 		require.NoError(t, err)
 		require.NotEmpty(t, vms)
+	})
+}
+
+func TestSessionReestablish(t *testing.T) {
+	simulator.Test(func(ctx context.Context, c *vim25.Client) {
+		// finder := find.NewFinder(c)
+		sm := session.NewManager(c)
+
+		moClient := &govmomi.Client{
+			Client:         c,
+			SessionManager: sm,
+		}
+		pw, _ := simulator.DefaultLogin.Password()
+		client := vcenterClient{
+			vimDriver: c,
+			cfg: &Config{
+				Username: simulator.DefaultLogin.Username(),
+				Password: pw,
+				Endpoint: fmt.Sprintf("%s://%s", c.URL().Scheme, c.URL().Host),
+				TLSClientSetting: configtls.TLSClientSetting{
+					Insecure: true,
+				},
+			},
+			moClient: moClient,
+		}
+		err := sm.Logout(ctx)
+		require.NoError(t, err)
+
+		err = client.EnsureConnection(ctx)
+		require.NoError(t, err)
 	})
 }

--- a/receiver/vcenterreceiver/client_test.go
+++ b/receiver/vcenterreceiver/client_test.go
@@ -71,9 +71,7 @@ func TestGetVMs(t *testing.T) {
 
 func TestSessionReestablish(t *testing.T) {
 	simulator.Test(func(ctx context.Context, c *vim25.Client) {
-		// finder := find.NewFinder(c)
 		sm := session.NewManager(c)
-
 		moClient := &govmomi.Client{
 			Client:         c,
 			SessionManager: sm,
@@ -94,10 +92,14 @@ func TestSessionReestablish(t *testing.T) {
 		err := sm.Logout(ctx)
 		require.NoError(t, err)
 
+		connected, err := client.moClient.SessionManager.SessionIsActive(ctx)
+		require.NoError(t, err)
+		require.False(t, connected)
+
 		err = client.EnsureConnection(ctx)
 		require.NoError(t, err)
 
-		connected, err := client.moClient.SessionManager.SessionIsActive(ctx)
+		connected, err = client.moClient.SessionManager.SessionIsActive(ctx)
 		require.NoError(t, err)
 		require.True(t, connected)
 	})

--- a/receiver/vcenterreceiver/client_test.go
+++ b/receiver/vcenterreceiver/client_test.go
@@ -96,5 +96,9 @@ func TestSessionReestablish(t *testing.T) {
 
 		err = client.EnsureConnection(ctx)
 		require.NoError(t, err)
+
+		connected, err := client.moClient.SessionManager.SessionIsActive(ctx)
+		require.NoError(t, err)
+		require.True(t, connected)
 	})
 }

--- a/unreleased/vcenter-refresh-sessions.yaml
+++ b/unreleased/vcenter-refresh-sessions.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: vcenterreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Re-establish sessions when running with particular TLS settings.
+
+# One or more tracking issues related to the change
+issues: [13447]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

